### PR TITLE
Fix a number of small local bugs

### PR DESCRIPTION
**Pull request by rneatherway**
### What does this PR do?

Fixes four minor local bugs. I found these at https://lgtm.com/projects/g/saltstack/salt/alerts/, and thought I would take a look at some of the shallower ones. There are some other pretty interesting ones there that I wasn't able to address myself. If you're interested you can also turn on PR integration and get warnings about these sorts of things ahead of merging (https://lgtm.com/projects/g/saltstack/salt/ci/).

### Tests written?

No

### CODE REVIEW

File: salt/modules/zabbix.py [hunk 1]
```diff
@@ -1726,8 +1726,6 @@ def run_query(method, params, **connection_args):
     ret = False
     try:
         if conn_args:
-            method = method
```
Comment (cachedout) [inline salt/modules/zabbix.py]: @mchugh19 Could you comment on this since you wrote this originally?

File: salt/returners/local_cache.py [hunk 2]
```diff
@@ -490,7 +490,7 @@ def save_reg(data):
     reg_dir = _reg_dir()
     regfile = os.path.join(reg_dir, 'register')
     try:
-        if not os.path.exists():
+        if not os.path.exists(reg_dir):
```
Comment (cachedout) [inline salt/returners/local_cache.py]: @techhat Can you comment on this since you wrote it originally?

File: salt/modules/x509.py [hunk 3]
```diff
@@ -1585,7 +1585,7 @@ def create_csr(path=None, text=False, **kwargs):
         log.warning(
             "OpenSSL no longer allows working with non-signed CSRs. A private_key must be specified. Attempting to use public_key as private_key")
 
-    if 'private_key' not in kwargs not in kwargs:
+    if 'private_key' not in kwargs:
```
Comment (cachedout) [inline salt/modules/x509.py]: @clinta Can you look at this change since you wrote it originally?

File: salt/ext/win_inet_pton.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ class sockaddr(ctypes.Structure):
     WSAStringToAddressA = ctypes.windll.ws2_32.WSAStringToAddressA
     WSAAddressToStringA = ctypes.windll.ws2_32.WSAAddressToStringA
 else:
-    def not_windows():
+    def not_windows(*args):
```
Comment (cachedout) [inline salt/ext/win_inet_pton.py]: @twangboy Can you comment on why this isn't accepting `*args` please?

File: salt/returners/local_cache.py (hunk 2 above)
Comment (techhat) [inline salt/returners/local_cache.py]: Looks like the fix is correct. Thanks @rneatherway.

File: salt/ext/win_inet_pton.py (hunk 4 above)
Comment (twangboy) [inline salt/ext/win_inet_pton.py]: Does it need to accept args? What was the behavior of this function before (what was the bug)?

Comment (rneatherway) [inline salt/ext/win_inet_pton.py]: WSAStringToAddressA and WSAAddressToStringA take arguments, so they are called with them below. I think it makes more sense for the body of this function to be executed and explain the problem than to throw a TypeError for being passed the wrong number of arguments. I found the issue here: https://lgtm.com/projects/g/saltstack/salt/snapshot/a3b59d268b5a7eafc74b7e1671dc64ce0159dea3/files/salt/ext/win_inet_pton.py#V44

File: salt/modules/x509.py (hunk 3 above)
Comment (clinta) [inline salt/modules/x509.py]: Fix looks good to me.

File: salt/ext/win_inet_pton.py (hunk 4 above)
Comment (cachedout) [inline salt/ext/win_inet_pton.py]: I agree with the analysis by @rneatherway. I believe this to be a latent bug.

File: salt/modules/zabbix.py (hunk 1 above)
Comment (mchugh19) [inline salt/modules/zabbix.py]: Yep, makes sense. Looks good. Thanks!

**Comment by cachedout** [comment]
Thanks for the quick responses everybody! Y'all are great. :]

**Pull request merged**